pkg/webhook/parser.go

```diff
@@ -362,20 +404,54 @@ func (Generator) RegisterMarkers(into *markers.Registry) error {
 	if err := into.Register(ConfigDefinition); err != nil {
 		return err
 	}
+	if err := into.Register(WebhookConfigDefinition); err != nil {
+		return err
+	}
 	into.AddHelp(ConfigDefinition, Config{}.Help())
+	into.AddHelp(WebhookConfigDefinition, Config{}.Help())
 	return nil
 }
 
 func (g Generator) Generate(ctx *genall.GenerationContext) error {
 	supportedWebhookVersions := supportedWebhookVersions()
 	mutatingCfgs := make(map[string][]admissionregv1.MutatingWebhook, len(supportedWebhookVersions))
 	validatingCfgs := make(map[string][]admissionregv1.ValidatingWebhook, len(supportedWebhookVersions))
+	mutatingWebhookCfgs := make(map[string]admissionregv1.MutatingWebhookConfiguration, len(supportedWebhookVersions))
+	validatingWebhookCfgs := make(map[string]admissionregv1.ValidatingWebhookConfiguration, len(supportedWebhookVersions))
+
 	for _, root := range ctx.Roots {
 		markerSet, err := markers.PackageMarkers(ctx.Collector, root)
 		if err != nil {
 			root.AddError(err)
 		}
 
+		webhookCfgs := markerSet[WebhookConfigDefinition.Name]
+		sort.SliceStable(webhookCfgs, func(i, j int) bool {
+			return webhookCfgs[i].(WebhookConfig).Name < webhookCfgs[j].(WebhookConfig).Name
+		})
+
+		for _, webhookCfg := range webhookCfgs {
+			webhookCfg := webhookCfg.(WebhookConfig)
+
+			if webhookCfg.Mutating {
+				w, err := webhookCfg.ToMutatingWebhookConfiguration()
+				if err != nil {
+					return err
+				}
+				for _, webhookVersion := range supportedWebhookVersions {
+					mutatingWebhookCfgs[webhookVersion] = w
```

Comment:
Last one wins instead of appending to an array because there'll only be one MutatingWebhookConfiguration and one ValidatingWebhookConfiguration per supported version.

Reply:
Q: This whole thing today only supports to generate one MutatingWebhookConfiguration and ValidatingWebhookConfiguration? (even before your change, I"m pretty sure just want to double check)

Reply:
yes, I think so